### PR TITLE
Include color components from Accessibility Developer Tools.

### LINKED ISSUE #1: Experiment with pulling in a11y-devtools `axs.color` functionality

### PR DESCRIPTION
**Pull request by ckundo**
- Copied from https://github.com/GoogleChrome/accessibility-developer-tools/blob/master/src/js/Color.js
- Use ES6 syntax and module pattern
- Pare down public API
- Add test coverage

The implementation is nearly the same as @alice's original in Accessibility
Developer Tools, except that we're now using ES6 syntax and modules and a
tighter public API.

[closes #1]

### CODE REVIEW

**Comment by ckundo** [comment]
@alice PTAL. I haven't tried pulling this into the Accessibility Developer Tools yet, but the tests are passing using the API I think the developer tools need. @dylanb also eager to hear your input on the API you'd want. cc @jdan

File: src/color.js [hunk 1]
```diff
@@ -0,0 +1,461 @@
+"use strict";
+
+/**
+ * color module
+ * @module color
```
Comment (ckundo) [inline src/color.js:5]: This is basically just a rewrite of https://github.com/GoogleChrome/accessibility-developer-tools/blob/master/src/js/Color.js in ES6 module syntax.

File: src/a11yCore.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+/**
+ * a11yCore module - used as a namespace.
+ * @module a11yCore
```
Comment (ckundo) [inline src/a11yCore.js:5]: This module acts as a wrapper and effectively namespace for the color module. There might be a better implementation, or a better API, but wanted to match what's in ADT as closely as possible initially.

File: src/color.js [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+
+/**
+ * Suggests alternative color suggestions to meet a given contrast ratio.
+ *
+ * @param {Color} bgColor - the background color.
+ * @param {Color} fgColor - the foreground color.
+ * @param {Object.<string, number>} desiredContrastRatios - A map of label to desired contrast ratio.
+ * @return {Object.<string, string>}
+ */
+export function suggestColors (bgColor, fgColor, desiredContrastRatios) {
```
Comment (jdan) [inline src/color.js:54]: I think it'd be ideal if the functions we exported as part of the API took human-readable strings, and we kept `parseColor` and our internal `Color` representation private.

Comment (jdan) [inline src/color.js:54]: Or, alternatively, we could accept `{ red, green, blue, alpha }` objects here - my comment should have been on `parseColor` above, which maybe should be private?

Comment (ckundo) [inline src/color.js:54]: @jdan I agree, I think `Color` and `parseColor` should be private. I only exposed methods that are being used by Accessibility Developer tools (e.g. https://github.com/GoogleChrome/accessibility-developer-tools/blob/master/src/js/AccessibilityUtils.js#L428), in the hopes of making the inclusion pretty seamless.

Comment (ckundo) [inline src/color.js:54]: @jdan what would you suggest as "human readable strings"? You mean the argument names?

Comment (jdan) [inline src/color.js:54]: @ckundo my original reaction was to just pass in hex codes here but meh it's not really better than an object with a few fields :) Just bike-shedding.

**Comment by jdan** [comment]
This is super rad. Do we need to add any attribution to the license since this uses code from ADT?

**Comment by ckundo** [comment]
> Do we need to add any attribution to the license since this uses code from ADT?

@alice can you comment on that?

**Comment by alice** [comment]
I believe the answer is "Add an AUTHORS file which lists Google Inc.", but I'll double check.

File: docs/index.js.html [hunk 4]
```diff
@@ -1,78 +0,0 @@
-<!DOCTYPE html>
```
Comment (alice) [inline docs/index.js.html]: Let's leave this in.

**Comment by alice** [comment]
Friendly lawyercat says "Check the commit history on color.js- add all authors (including non-Googlers) who ever touched the file to your AUTHORS, and preserve the headers."

**Comment by alice** [comment]
Oh, but for my authorship you need only say "Google Inc.".

**Comment by alice** [comment]
Sorry for being so slow on this! Looking now and trying to get my head around ES6 syntax. How do I run the tests?

**Comment by ckundo** [comment]
I think you can just run `gulp`. If not try `npm test`.

**Comment by ckundo** [comment]
@alice I added an AUTHORS file and squashed the commits. Can you please take another look?

**Comment by alice** [comment]
Ok I had to think about this for a year, but this LGTM, let's do it 😀

**Pull request merged**